### ISSUE TITLE
What is the recommended way of searching for tasks in a particular page?

### ISSUE THREAD

**Issue by evan54** [zim-desktop-wiki/zim-desktop-wiki#1307]
This is my attempt to write a first custom tool. The idea is that it will run through the journal pages and go through and see what tasks are outstanding / not outstanding

It will then copy any not outstanding tasks to today's journal page into a new section.

I know I can do all of this by writing my own regexp expressions but I was wondering if there is a recommended way that uses existing parsing functions for example to identify if a line is a completed task or not, if a line is a header or not etc.

I tried to look at the tasklist plugin code but I have to say that seemed a lot more confusing than what I had expected, I also saw there was SQL code involved which was a bit surprising as I thought everything is stored in the text files rather than smth else - am I missing something?

To sum up my questions are:
* If I'm writing a parsing tool for tasks and edits the journal pages should I just use my own regexp or is there some built in class  / function I should be using?
* Why is there SQL code for the tasklist plugin and should I worry about that if I'm effectively going to be creating tasks through code in Journal pages?

thank you

**Comment by jaap-karssenberg** [comment]
There is an index that caches content for fast lookups. The SQL is used to
query this cache.

In the tasklist "indexer" module there is a "View" class that exposes an
API around this. THis you can use to list open tasks without touching any
of the SQL directly.

-- Jaap

**Comment by evan54** [comment]
but modifying the SQL won't change the actual task at hand (eg mark it complete) right?

My initial attempt at this is here where I'm actually just reading and writing the files of interest directly and seems to work.

https://github.com/evan54/zimplugin/blob/main/main.py

I'll check out the index class in case that helps, but if I can't modify actual files through it then would probably not need it.

**Comment by jaap-karssenberg** [comment]
No, the indexer will only help finding the tasks. To change then you need
to change the pages where they occur.

Op di 3 nov. 2020 19:58 schreef evan54 <notifications@github.com>:
